libs/blocks/merch/merch.js

```diff
@@ -100,6 +100,14 @@ window.tacocat.loadPromise = new Promise((resolve) => {
 
   loadScript(literalScriptUrl)
     .catch(() => ({})) /* ignore if literals fail */
+    .then(() => {
+      // load custom elements polyfill for Safari only.
+      if (navigator.userAgent.indexOf('Safari') >= 0 && navigator.userAgent.indexOf('Chrome') < 0) {
```

Comment:
Agree, MDN discourages this too: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent

Reply:
@overmyheadandbody I know customElements api is well supported in Safari.
What is not supported is **_Customized built-in elements_** [1] which is a lightweight version of web components.
Apparently Safari Web kit teams doesn't want to implement it in near feature.
AFAIK there is no API way to check whether this feature is supported by a browser other than checking https://caniuse.com/?search=Custom%20Elements%20V1.

[1] - https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements#customized_built-in_elements

Reply:
It would be better to directly detect for `is` support in case safari adds it in the future:
```
const supportsCustomizedEl = () => {
  const BR = () => Reflect.construct(HTMLBRElement, [], BR);
  BR.prototype = HTMLLIElement.prototype;
  const is = 'extends-br';
  return document.createElement('br', { is }).outerHTML.includes(is);
};
```
From: https://github.com/ungap/custom-elements/blob/master/index.js#L370

Reply:
Another option might be
```
let supports = false;
document.createElement('div', {
  get is() {
    supports = true; 
  }
});
```
taken from https://stackoverflow.com/a/72345670

Reply:
Thanks for the propositions.
Why we should run this code for the majority of the browsers, especially during LH test with Chrome?
If one day Safari implements it, we will eventually get to know it and can update Milo with a PR.
A `navigator.userAgent` based check seems to be more than enough.

Reply:
A quick test on my local shows that @overmyheadandbody version is faster than `userAgent` and my version.  I get around 7ms for `supports` and 11ms for `userAgent`
I think that feature detection is always preferred to browser detection.

Reply:
+1 for feature detection over browser one.

Reply:
Even if Safari 17 would introduce full support for the feature you're looking for, you won't be able to make the switch immediately, as there will be plenty of users on older versions, so you'll be stuck with the user agent check for an additional 1-2 years. Whereas using feature detection will always ensure that the polyfills are loaded only if truly needed.

Reply:
Very good point @overmyheadandbody thanks.